contracts/ProtocolReserve/ProtocolShareReserve.sol

```diff
@@ -1,29 +1,93 @@
 // SPDX-License-Identifier: BSD-3-Clause
 pragma solidity 0.8.13;
 
-import { Ownable2StepUpgradeable } from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
 import { SafeERC20Upgradeable, IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { AccessControlledV8 } from "@venusprotocol/governance-contracts/contracts/Governance/AccessControlledV8.sol";
 
-import { ExponentialNoError } from "../Utils/ExponentialNoError.sol";
-import { IRiskFund } from "../Interfaces/IRiskFund.sol";
-import { ReserveHelpers } from "../Helpers/ReserveHelpers.sol";
 import { IProtocolShareReserve } from "../Interfaces/IProtocolShareReserve.sol";
+import { ComptrollerInterface } from "../Interfaces/ComptrollerInterface.sol";
+import { PoolRegistryInterface } from "../Interfaces/PoolRegistryInterface.sol";
+import { IPrime } from "../Interfaces/IPrime.sol";
+import { IVToken } from "../Interfaces/IVToken.sol";
+import { IIncomeDestination } from "../Interfaces/IIncomeDestination.sol";
 
-contract ProtocolShareReserve is Ownable2StepUpgradeable, ExponentialNoError, ReserveHelpers, IProtocolShareReserve {
+contract ProtocolShareReserve is AccessControlledV8, IProtocolShareReserve {
     using SafeERC20Upgradeable for IERC20Upgradeable;
 
-    address private protocolIncome;
-    address private riskFund;
-    // Percentage of funds not sent to the RiskFund contract when the funds are released, following the project Tokenomics
-    uint256 private constant protocolSharePercentage = 70;
-    uint256 private constant baseUnit = 100;
+    /// @notice protocol income is categorized into two schemas.
+    /// The first schema is for spread income from prime markets in core protocol
+    /// The second schema is for all other sources and types of income
+    enum Schema {
+        ONE,
+        TWO
+    }
+
+    struct DistributionConfig {
+        Schema schema;
+        /// @dev percenatge is represented without any scale
+        uint256 percentage;
+        address destination;
+    }
+
+    /// @notice address of Prime contract
+    address public prime;
+
+    /// @notice address of pool registry contract
+    address public poolRegistry;
+
+    /// @notice address of core pool comptroller contract
+    address public CORE_POOL_COMPTROLLER;
+
+    uint256 private constant MAX_PERCENT = 100;
+
+    /// @notice comptroller => asset => schema => balance
+    mapping(address => mapping(address => mapping(Schema => uint256))) public assetsReserves;
+
+    /// @notice asset => balance
+    mapping(address => uint256) private totalAssetReserve;
```

Comment:
```suggestion
    /// @dev asset => balance
    mapping(address => uint256) private totalAssetReserve;
```
I think private variables netspac is `@dev` @coreyar can you please confirm here